### PR TITLE
fix(medusa-plugin-sendgrid): Inject `GiftCardService` to eliminate `undefined` error

### PR DESCRIPTION
**Pull request by RegisWeduLabs**
This PR fixes an issue where gift card created from the admin portal wouldn't send using sendgrid.

Also - there is no order attached to the gift card when it's created trough the admin portal - so I revoved the return statement if there is no order and adjusted the data sent to the template based on what was there - I updated the documentation to reflect the "display_value".

### CODE REVIEW

**Comment by RegisHubelia** [comment]
Can someone look at this PR and let me know if any changes is required?

Thanks

File: packages/medusa-plugin-sendgrid/src/services/sendgrid.js [hunk 1]
```diff
@@ -579,20 +581,19 @@ class SendGridService extends NotificationService {
...
+    // if the gift card has no order, we still return the email associated to the gift card (created from admin panel).
+    const email = giftCard.order ? giftCard.order.email : giftCard.metadata.email;
 
     return {
       ...giftCard,
       locale,
-      email: giftCard.order.email,
-      display_value: giftCard.value * (1 + taxRate),
+      email,
+      display_value: (giftCard.value * (1 + taxRate) / 100).toFixed(2),
```
Comment (olivermrbl) [inline packages/medusa-plugin-sendgrid/src/services/sendgrid.js:595]: todo: Use util to cover all currency cases
```suggestion
      display_value: `${this.humanPrice_((giftCard.value * 1+ taxRate), giftCard.region.currency_code)} ${giftCard.region.currency_code}`,
```

File: packages/medusa-plugin-sendgrid/src/services/sendgrid.js [hunk 2]
```diff
@@ -579,20 +581,19 @@ class SendGridService extends NotificationService {
     const giftCard = await this.giftCardService_.retrieve(id, {
       relations: ["region", "order"],
     })
-
-    if (!giftCard.order) {
-      return
-    }
-
+    // Should this use the custom Tax Provider if any?
```
Comment (olivermrbl) [inline packages/medusa-plugin-sendgrid/src/services/sendgrid.js:584]: todo: Remove comment

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Thanks for your contribution. I've added a couple of todo. Let me know if something is unclear :)

**Comment by RegisHubelia** [comment]
Perfect  - will do and send the commit shortly.

**Comment by daysgobye** [comment]
Thank you for catching this. Glad someone is working on it cant wait for it to be merged.

File: packages/medusa-plugin-sendgrid/src/services/sendgrid.js (hunk 2 above)
Comment (RegisHubelia) [reply packages/medusa-plugin-sendgrid/src/services/sendgrid.js]: Done!

**Comment by RegisHubelia** [comment]
> Also, please add a [changeset](https://github.com/changesets/changesets) bumping the Sendgrid plugin with the following message:
> 
> ```
> fix(medusa-plugin-sendgrid): Inject GiftCardService to eliminate undefined error
> ```

Done

**Comment by RegisHubelia** [comment]
@olivermrbl - Let me know if I need to do anything else?

**Comment by olivermrbl** [comment]
@RegisHubelia Your branch is out-of-date with the base. Can I get you to update?

**Comment by RegisHubelia** [comment]
@olivermrbl 
> @RegisHubelia Your branch is out-of-date with the base. Can I get you to update?

Just did!

**Review by olivermrbl — APPROVED** [review]
LGTM! Will be part of next release :)

**Comment by daysgobye** [comment]
what is eta on this being merged?

**Comment by olivermrbl** [comment]
@RegisHubelia Can I get you to update your branch with latest from `develop`? Then I'll push to get it merged today :)

**Comment by RegisHubelia** [comment]
> @RegisHubelia Can I get you to update your branch with latest from `develop`? Then I'll push to get it merged today :)

Done!

**Pull request merged**